pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -313,6 +318,48 @@ def original_device(self):
 
         return TransformedDevice(original_device, self._transform)
 
+    def _batch_transform(self, original_batch, targs, tkwargs):
+        """Apply the transform on a batch of tapes"""
+        execution_tapes = []
+        batch_fns = []
+        tape_counts = []
+
+        for t in original_batch:
+            # Preprocess the tapes by applying batch transforms
+            # to each tape, and storing corresponding tapes
+            # for execution, processing functions, and list of tape lengths.
+            new_tapes, fn = self(t, *targs, **tkwargs)
+            execution_tapes.extend(new_tapes)
+            batch_fns.append(fn)
+            tape_counts.append(len(new_tapes))
+
+        def processing_fn(res: ResultBatch) -> ResultBatch:
+            """Applies a batch of post-processing functions to results.
+
+            Args:
+                res (ResultBatch): the results of executing a batch of circuits
+
+            Returns:
+                ResultBatch : results that have undergone classical post processing
+
+            Closure variables:
+                tape_counts: the number of tapes outputted from each application of the transform
+                batch_fns: the post processing functions to apply to each sub-batch
+
+            """
+            count = 0
+            final_results = []
+
+            for f, s in zip(batch_fns, tape_counts):
+                # apply any batch transform post-processing
+                new_res = f(res[count : count + s])
+                final_results.append(new_res)
+                count += s
+
+            return final_results
+
+        return execution_tapes, processing_fn
```

Comment:
Actually, mind casting both `final_results` and `execution_tapes` back to tuples before returning? Tuples are both more performant and protected from accidental mutation.

Basically lists get optimized for the ability to be quickly mutated.  This means they take up a lot more space and have an extra level of indirection on lookup.